### PR TITLE
ARROW-18016: [CI] Add sccache to more jobs

### PR DESCRIPTION
**Pull request by assignUser**

### CODE REVIEW

**Comment by assignUser** [comment]
@github-actions crossbow submit -g nightly

**Comment by pitrou** [comment]
By the way, what is the current disk footprint of our sccache?

**Comment by assignUser** [comment]
@github-actions crossbow submit -g nightly

**Comment by assignUser** [comment]
@github-actions crossbow submit -g nightly

**Comment by assignUser** [comment]
@raulcd @kou could you review this? There are still a few jobs we can add sccache but I will do those in a follow up to keep this reviewable.

File: ci/docker/conda-integration.dockerfile [hunk 1]
```diff
@@ -36,7 +36,8 @@ RUN mamba install -q -y \
         maven=${maven} \
         nodejs=${node} \
         yarn \
-        openjdk=${jdk} && \
+        openjdk=${jdk} \
+        sccache && \
```
Comment (kou) [inline ci/docker/conda-integration.dockerfile:40]: Could you sort this list in alphabetical order?

File: dev/tasks/macros.jinja [hunk 2]
```diff
@@ -383,3 +383,16 @@ on:
...
 {% endmacro %}
+
+{% macro azure_set_sccache_envvars(sccache_key_prefix = "sccache") %}
+  {% set sccache_vars =  {
+            "AWS_SECRET_ACCESS_KEY": '$(AWS_SECRET_ACCESS_KEY)',
+            "AWS_ACCESS_KEY_ID": '$(AWS_ACCESS_KEY_ID)',
+            "SCCACHE_BUCKET": '$(SCCACHE_BUCKET)',
+            "SCCACHE_S3_KEY_PREFIX": sccache_key_prefix 
+          }
+ %}
```
Comment (kou) [inline dev/tasks/macros.jinja:394]:
```suggestion
  %}
```

File: dev/tasks/macros.jinja [hunk 3]
```diff
@@ -383,3 +383,16 @@ on:
   {{ key }}: "{{ value }}"
   {% endfor %}
 {% endmacro %}
+
+{% macro azure_set_sccache_envvars(sccache_key_prefix = "sccache") %}
+  {% set sccache_vars =  {
```
Comment (kou) [inline dev/tasks/macros.jinja:388]:
```suggestion
  {% set sccache_vars = {
```

File: dev/tasks/macros.jinja [hunk 4]
```diff
@@ -383,3 +383,16 @@ on:
...
+            "AWS_SECRET_ACCESS_KEY": '$(AWS_SECRET_ACCESS_KEY)',
+            "AWS_ACCESS_KEY_ID": '$(AWS_ACCESS_KEY_ID)',
```
Comment (kou) [inline dev/tasks/macros.jinja:390]: Could you sort this list in alphabetical order?

File: dev/tasks/r/github.macos.brew.yml [hunk 5]
```diff
@@ -27,17 +27,20 @@ jobs:
       {{ macros.github_checkout_arrow()|indent }}
 
       {{ macros.configure_homebrew_arrow(formula)|indent }}
-
+
```
Comment (kou) [inline dev/tasks/r/github.macos.brew.yml:30]:
```suggestion

```

**Review by kou — COMMENTED** [review]
How to confirm that cache is used expectedly? There are many jobs...

**Comment by assignUser** [comment]
@github-action crossbow submit verify-rc-source-cpp-linux-conda-latest-amd64 test-alpine-linux-cpp homebrew-r-autobrew test-r-install-local test-r-linux-as-cran test-r-versions

**Comment by assignUser** [comment]
@kou the only way is to manually check in the logs :see_no_evil:  but I did that and noted the jobs that are still not using sccache. The builds above had errors (missing curl)/I fixed/were still missing envvars so I am now only testing those. all other failures are unrelated.

**Comment by assignUser** [comment]
The R failures are unrelated test failure, I had to change the install script to work with alpine's busybox

**Comment by assignUser** [comment]
@github-actions crossbow submit test-alpine-linux-cpp test-ubuntu-20.04-cpp

**Comment by assignUser** [comment]
rebased to fix go CI issue

**Comment by assignUser** [comment]
@github-actions crossbow submit -g nightly

**Comment by assignUser** [comment]
- python-sdist 
  - was still missing curl, now fixed 
- test-r-versions 
  - fails due to [ARROW-18174](https://issues.apache.org/jira/browse/ARROW-18174)
- verify-rc-source-integration-linux-ubuntu-18.04-amd64 
- verify-rc-source-js-linux-ubuntu-18.04-amd64 
  - fail due to wrong [nodejs version](https://github.com/ursacomputing/crossbow/actions/runs/3345844532/jobs/5541889144) (18 is not supported)  

Will update this when more results are available.

**Comment by assignUser** [comment]
@github-actions crossbow submit r-binary-packages python-sdist

**Comment by assignUser** [comment]
The r binary failure is also caused by ARROW-18174, the problem caused by busybox being installed was solved.

**Comment by assignUser** [comment]
@github-actions crossbow submit test-alpine-linux-cpp ubuntu-cpp

**Comment by kou** [comment]
>     * python-sdist
>       
>       * was still missing curl, now fixed

Why do we need sscache for this? It seems that we don't compile C/C++ code for this.
 
BTW, could you list jobs that should be sscache enabled? I'll confirm results of them.

**Comment by assignUser** [comment]
This PR is to widespread to review properly. I am going to open a new PR with reduced scope.

**Pull request closed without merging**